### PR TITLE
Redux for gem2s

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-949

#### Link to staging deployment URL 
https://ui-martinfosco-ui260-api117-w.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/api/pull/117

#### Anything else the reviewers should know about the changes here
Most of the actual work for displaying was done as part of https://biomage.atlassian.net/browse/BIOMAGE-751.

In this PR I focused on some minor refactoring and renaming to have the redux store more accurately reflect what information the status is holding.

I also added:
- Take one extra value from backend status updates which specifies what type of status message is being sent (either pipeline or gem2s).

- Merge status instead of overwrite (but overwriting arrays) in order to not have the pipeline updates cause the gem2s status to disappear, or viceversa.

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/utils/experimentUpdatesHandler.js [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+    Object.entries(output.plotData).forEach(([plotUuid, plotData]) => {
+      dispatch(updatePlotData(plotUuid, plotData));
+    });
 
-  dispatch(updatePipelineStatus(experimentId, status));
-  dispatch(updateProcessingSettings(experimentId, input.taskName, processingConfigUpdate));
+    // This is temporary, should be taken out eventually as it will be handled
+    // once we clean up the redux store (which is the correct action to take here)
+    dispatch(loadEmbedding(experimentId, 'umap'));
+    dispatch(loadEmbedding(experimentId, 'tsne'));
```
Comment (aerlaut) [inline src/utils/experimentUpdatesHandler.js:40]: Do you know why this should be dispatched? Is it still necessary? Or can we remove it?

**Comment by aerlaut** [comment]
How did you test locally? Did you manage to trigger changes in the loading UI?

File: src/utils/experimentUpdatesHandler.js (hunk 1 above)
Comment (cosa65) [reply src/utils/experimentUpdatesHandler.js:40]: I think we had a conversation about this some weeks ago, wasn't it the thing that for some reason when removed the pipeline broke down?

**Comment by cosa65** [comment]
> How did you test locally? Did you manage to trigger changes in the loading UI?

I tested not really on the socket, but rather by adding fake gem2s data in backend-status.js so that the endpoint `pipelines#get` returns it.

File: src/utils/experimentUpdatesHandler.js (hunk 1 above)
Comment (aerlaut) [reply src/utils/experimentUpdatesHandler.js:40]: Oh right!

**Review by aerlaut — APPROVED** [review]
LGTM

**Comment by aerlaut** [comment]
Arrival of the message from GEM2S will not yet update the UI because `onGEM2SUpdate` is not yet implemented. Once we know the payload. This should be implemented in 947.

**Pull request merged**